master-firmware/src/udp_topic_injector.c

```diff
@@ -0,0 +1,43 @@
+#include <ch.h>
+#include <lwip/api.h>
+#include "udp_topic_injector.h"
+#include "msgbus_protobuf.h"
+#include "main.h"
+
+static void udp_topic_injector_thd(void* p)
+{
+    (void)p;
+
+    struct netconn* conn;
+
+    chThdSetPriority(HIGHPRIO);
+    chRegSetThreadName(__FUNCTION__);
+
+    /* Listens to UDP on port 10'000 */
+    conn = netconn_new(NETCONN_UDP);
+    chDbgAssert(conn != NULL, "Cannot create a connection object");
+    netconn_bind(conn, IPADDR_ANY, 10000);
+
+    while (true) {
+        struct netbuf* buf;
+
+        /* Read a datagram */
+        if (netconn_recv(conn, &buf) != ERR_OK) {
+            continue;
+        }
+
+        /* TODO: Better than a static object ? */
+        static uint8_t msg[1024];
+        netbuf_copy(buf, msg, sizeof(msg));
+
+        messagebus_inject_encoded_message(&bus, msg, netbuf_len(buf));
```

Comment:
as if it was meant to work with UDP :stuck_out_tongue:

Reply:
? What do you mean ?

Reply:
![thatwasajoke](https://user-images.githubusercontent.com/6131403/73594856-7d2c3b00-4512-11ea-8704-e59c6e138d74.gif)